### PR TITLE
Improvements on event logging for very active contracts

### PR DESCRIPTION
**Pull request by alexvandesande**
Optimizations for a large amount of events per block in one watched account. Lots of users watching the Dao were affected by it.

### CODE REVIEW

File: app/client/lib/ethereum/observeEvents.js [hunk 1]
```diff
@@ -12,15 +14,31 @@ observeEvents = function(){
...
     collectionObservers[collectionObservers.length] = Events.find({}).observe({
         /**
-        This will observe the transactions creation and create watchers for outgoing transactions, to see when they are mined.
+        This will observe when events are added and link it to the custom contract.
 
         @method added
         */
         added: function(newDocument) {
+            
+            // This creates a temporary cache for the contracts, to reduce the amount of db reads
```
Comment (bas-vk) [inline app/client/lib/ethereum/observeEvents.js]: I believe that you are trying to reduce the amount of writes/updates instead of reads?

File: app/client/lib/ethereum/observeEvents.js [hunk 2]
```diff
@@ -12,15 +14,31 @@ observeEvents = function(){
...
 
         @method added
         */
         added: function(newDocument) {
+            
+            // This creates a temporary cache for the contracts, to reduce the amount of db reads            
+            var customContract = customContractsCache[newDocument.address.toLowerCase()];
+
+            if (typeof customContract == 'undefined') {
+                var customContract = CustomContracts.findOne({address: newDocument.address.toLowerCase()});
```
Comment (bas-vk) [inline app/client/lib/ethereum/observeEvents.js]: Is it possible to move line 39 into this if statement?
Currently it will always execute this line if the `customContract` isn't in the cache which will only happen after at least 1 unknown event is raised. Until then each time `findOne` is executed.

File: app/client/lib/ethereum/observeBlocks.js [hunk 3]
```diff
@@ -109,4 +110,5 @@ observeLatestBlocks = function(){
     peerCountIntervalId = setInterval(function() {
         getPeerCount()
     }, 1000);
+    console.timeEnd('observeLatestBlocks');
```
Comment (hiddentao) [inline app/client/lib/ethereum/observeBlocks.js]: Don't understand why this section is being timed since some of the code is asynchronously executed anyway. Surely the timing code should go into `updateBalances()` instead?

File: app/client/lib/ethereum/observeEvents.js (hunk 1 above)
Comment (hiddentao) [inline app/client/lib/ethereum/observeEvents.js]: From what I can see it looks like `update()` is only called if the `_id` is not already stored inside `contractEvents`. And this check is performed by loading the custom contract once using `findOne()` and cacheing it. So I agree, it's limiting the no. of db writes.

File: app/client/lib/ethereum/observeCustomContracts.js [hunk 4]
```diff
@@ -33,8 +33,6 @@ var addLogWatching = function(newDocument){
         if(!error) {
             var id = Helpers.makeId('log', web3.sha3(log.logIndex + 'x' + log.transactionHash + 'x' + log.blockHash));
 
-            Helpers.eventLogs(log);
```
Comment (hiddentao) [inline app/client/lib/ethereum/observeCustomContracts.js]: Why is this removed?

Comment (alexvandesande) [inline app/client/lib/ethereum/observeCustomContracts.js]: Having a log for every event detected would flood the console.log when observing a very active contract

File: app/client/lib/ethereum/observeEvents.js (hunk 1 above)
Comment (alexvandesande) [inline app/client/lib/ethereum/observeEvents.js]: I'm also trying to limit the amount of reads
